tests/support/peerManager.ts

```diff
@@ -89,6 +92,14 @@ export class RadiclePeer {
     this.#outputLog = props.logFile;
   }
 
+  public waitForEvent(searchEvent: NodeEvent, timeoutInMs: number) {
+    const start = new Date().getTime();
+
+    while (new Date().getTime() - start > timeoutInMs) {
+      this.#eventRecords.find(event => lodash.isEqual(searchEvent, event));
```

Comment:
I'd add a 100ms sleep here, otherwise this will load the CPU 100%.

Reply:
Sure

Reply:
e4c2861